pub/spec/dependabot/pub/update_checker_spec.rb

```diff
@@ -465,6 +468,94 @@
+          dependency_name: dependency_name,
+          package_manager: "pub",
+          vulnerable_versions: ["<3.0.0"]
+        )
+      ]
+    end
+
+    context "when a newer non-vulnerable version is available" do
+      # TODO: Implement https://github.com/dependabot/dependabot-core/issues/5391, then flip "highest" to "lowest"
+      it "updates to the highest non-vulnerable version" do
+        is_expected.to eq(Gem::Version.new("3.1.0"))
+      end
+    end
+
+    # TODO should it update indirect deps for security vulnerabilities? I assume Pub has these? example in go_modules/update_checker_spec
+
+    context "when the current version is not newest but also not vulnerable" do
+      let(:dependency_version) { "3.0.0" } # 3.1.0 is latest
+      it "raises an error " do
+        expect { lowest_resolvable_security_fix_version.to }.to raise_error(RuntimeError) do |error|
+          expect(error.message).to eq("Dependency not vulnerable!")
+        end
+      end
+    end
+  end
+
+  describe "#lowest_security_fix_version" do
+    subject(:lowest_security_fix_version) { checker.lowest_security_fix_version }
+    let(:dependency_name) { "retry" }
+
+    # TODO: Implement https://github.com/dependabot/dependabot-core/issues/5391, then flip "highest" to "lowest"
+    it "finds the highest available non-vulnerable version" do
+      is_expected.to eq(Gem::Version.new("3.1.0"))
+    end
+
+    context "with a security vulnerability on older versions" do
+      let(:security_advisories) do
+        [
+          Dependabot::SecurityAdvisory.new(
+            dependency_name: dependency_name,
+            package_manager: "pub",
+            vulnerable_versions: ["< 3.0.0"]
+          )
+        ]
+      end
+
+      # TODO: Implement https://github.com/dependabot/dependabot-core/issues/5391, then flip "highest" to "lowest"
+      it "finds the highest available non-vulnerable version" do
+        is_expected.to eq(Gem::Version.new("3.1.0"))
+      end
+
+      # TODO: Implement https://github.com/dependabot/dependabot-core/issues/5391, then this should return the "lowest" that isn't ignored
+      context "when the latest version is being ignored" do
+        let(:ignored_versions) { ["> 3.0.0"] }
+        # TODO should I even set raise_on_ignored here? or is it not supposed to be set?
+        let(:raise_on_ignored) { true } # TODO this has no effect because version_unless_ignored doesn't raise like I think it should?
+        pending it "raises an error" do
+          expect { subject }.to raise_error(Dependabot::AllVersionsIgnored)
+        end
+      end
+
+      # it "returns nil for git versions" # tested elsewhere under `context "With a git dependency"`
+
+      # TODO how are pre-releases supposed to be handled? pub readme says if on pre-release, bumped to next pre-release, else not...
+      # check with the pub guys re: how to test... I see requires_old_beta fixture, not sure how to use it though?
```

Comment:
One final question for you @jonasfj / @sigurdm ☝️ 

I can dig more in the code, but hoping that since you wrote this and have been super responsive you might know the answer off top of your head and save me some time.

Also happy to have a short video call to go over each of these questions if that'd be more time efficient.